### ISSUE TITLE
Dataset splits and reproductibility

### ISSUE THREAD

**Issue by clementw168** [LUOyk1999/tunedGNN#14]
Hey, 

Thank you for your work! It is an amazing contribution for the community. 

I have a question about the splits for Cora, Citeseer, and Pubmed. I had trouble to reproduce your results with my own code. 

The code to run the benchmarking is the following: 

```
python main.py --gnn gcn --dataset cora --lr 0.001 --local_layers 3  --hidden_channels 512 --weight_decay 5e-4 --dropout 0.7 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn sage --dataset cora --lr 0.001 --local_layers 3  --hidden_channels 256 --weight_decay 5e-4 --dropout 0.7 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn gat --dataset cora --lr 0.001 --local_layers 3  --hidden_channels 512 --weight_decay 5e-4 --dropout 0.2 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5 --res

python main.py --gnn gcn --dataset citeseer --lr 0.001 --local_layers 2 --hidden_channels 512 --weight_decay 0.01 --dropout 0.5 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn sage --dataset citeseer --lr 0.001 --local_layers 3 --hidden_channels 512 --weight_decay 0.01 --dropout 0.2 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn gat --dataset citeseer --lr 0.001 --local_layers 3 --hidden_channels 256 --weight_decay 0.01 --dropout 0.5 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5 --res

python main.py --gnn gcn --dataset pubmed --lr 0.005 --local_layers 2 --hidden_channels 256 --weight_decay 5e-4 --dropout 0.7 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn sage --dataset pubmed --lr 0.005 --local_layers 4 --hidden_channels 512 --weight_decay 5e-4 --dropout 0.7 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5
python main.py --gnn gat --dataset pubmed --lr 0.01 --local_layers 2 --hidden_channels 512 --weight_decay 5e-4 --dropout 0.5 --rand_split_class --valid_num 500 --test_num 1000 --seed 123 --device $1 --runs 5

```

Why do you use random splits (`--rand_split_class`) for these datasets ? When we change the seed here, the seed from 123 to 22 on Cora GCN for instance, the results on the test set vary a lot, 85% -> 81% with ~0.5 std. When digging a bit more, I found out that you do the random split before the for loop for each run. I think you it would be better to do the splits inside of the for loop when we use random splits. What do you think?

Also, I find it a bit disturbing to monitor the metrics on the test set during the whole training and having the highest test accuracy. Does it not bias hyperparameters tuning?

**Comment by LUOyk1999** [comment]
Thank you for your question!

We use random splits on these datasets because this was the strategy adopted in prior work, and we followed it for consistency. That said, we also experimented with other splitting strategies and found the results to be consistent across them.

From our experience, when you change the random seed, it is often necessary to slightly adjust certain hyperparameters (e.g., dropout, number of layers, or hidden dimension). With such minor adjustments, you can usually recover similar results.
